actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -240,6 +260,12 @@ def signed_or_encrypted
         def upgrade_legacy_signed_cookies?
           request.secret_token.present? && request.secret_key_base.present?
         end
+
+        def upgrade_legacy_hmac_aes_cbc_cookies?
+          request.secret_key_base.present?                 &&
+            request.use_authenticated_cookie_encryption?   &&
+            request.use_authenticated_cookie_encryption?
```

Comment:
Checking `secret_key_base` seemed a bit unnecessary and I do think it's likely safe to drop at this point.

I'll also rework my commits soon and drop the `upgrade_legacy_hmac_aes_cbc_cookies` flag and revert the `upgrade_legacy_hmac_aes_cbc_cookies?` method to just check if the salt values are `present?`. I'll also update the above mention changes to the various guides. At that point, I think we'll be close to ready for this PR. Thanks again!!